packages/runtime/container-runtime/src/gcSweepReadyUsageDetection.ts

```
@@ -0,0 +1,130 @@
+            } catch (error) {
+                this.localStorage = noopStorage;
+            }
+        }
+
+        if (this.localStorage === noopStorage) {
+            // This means the Blackout Period logic will not work.
+            this.mc.logger.sendTelemetryEvent({ eventName: "SweepReadyUsageDetectionHandlerNoOpStorage" });
+        }
+    }
+
+    /**
+      * If SweepReady Usage Detection is enabled, close the main container.
+      * If the "Blackout Period" is set, don't close the container more than once in that period.
+      *
+      * Once Sweep is fully implemented, this will be removed since the objects will be gone
+      * and errors will arise elsewhere in the runtime
+     */
+    public usageDetectedInMainContainer(errorProps: ITelemetryProperties) {
+        if (!sweepReadyUsageDetectionSetting.read(this.mc.config).mainContainer) {
+            return;
+        }
+
+        const pastClosuresMap: Record<string, { lastCloseTime: number; } | undefined> = (() => {
+            try {
+                const rawValue = this.localStorage.getItem(closuresStorageKey);
+                return rawValue === null
+                    ? {}
+                    : JSON.parse(rawValue) as Record<string, { lastCloseTime: number; } | undefined>;
+            } catch (e) {
+                return {};
+            }
+        })();
+
+        const lastCloseTime = pastClosuresMap[this.uniqueContainerKey]?.lastCloseTime;
+        const blackoutPeriodDays = this.mc.config.getNumber(blackoutPeriodDaysKey);
+
+        const error = new SweepReadyUsageError(
+            "SweepReady object used in Non-Summarizer Client",
+            { errorDetails: JSON.stringify({ ...errorProps, lastCloseTime, blackoutPeriodDays }) },
+        );
+
+        // Should close if...
+        const shouldClose = lastCloseTime === undefined // ...We've not closed before, or...
+            || blackoutPeriodDays === undefined // ...there's no blackout duration set, or...
```

What's blackoutPeriodDays?

---

A little more details on how this will be used will be helpful :-)

---

Take a look at latest iteration.  Also this function's doc comments mentions the term.  Where would be the best place to explain how this works?

---

If I understand correctly, blackoutPeriodDays is the number of days this error gets ignored if it's not the first time it occurs.

---

Yep.  Will be clearer after this little refactor I think.